pyglet/gui/frame.py

```diff
@@ -87,49 +88,84 @@ def remove_widget(self, widget):
 
     def on_mouse_press(self, x, y, buttons, modifiers):
         """Pass the event to any widgets within range of the mouse"""
+
+        if self._active_entry: 
+            self._active_entry.on_mouse_press(x, y, buttons, modifiers)
+            self._active_entry = self._active_entry if self._active_entry._focus else None
+        
+
         for widget in self._cells.get(self._hash(x, y), set()):
-            widget.on_mouse_press(x, y, buttons, modifiers)
+            if type(widget).__name__ == 'TextEntry':
+                if self._active_entry == widget: continue
+                widget.on_mouse_press(x, y, buttons, modifiers)
+                self._active_entry = widget if widget._check_hit(x, y) else self._active_entry
+
+            else: widget.on_mouse_press(x, y, buttons, modifiers)
             self._active_widgets.add(widget)
 
+
     def on_mouse_release(self, x, y, buttons, modifiers):
         """Pass the event to any widgets that are currently active"""
+        if self._active_entry: self._active_entry.on_mouse_release(x, y, buttons, modifiers)
+
         for widget in self._active_widgets:
             widget.on_mouse_release(x, y, buttons, modifiers)
         self._active_widgets.clear()
 
     def on_mouse_drag(self, x, y, dx, dy, buttons, modifiers):
         """Pass the event to any widgets that are currently active"""
+        if self._active_entry: self._active_entry.on_mouse_drag(x, y, dx, dy, buttons, modifiers)
+
         for widget in self._active_widgets:
+            if widget == self._active_entry: continue
```

Comment:
Can you explain this special case in `mouse_drag` but not in `mouse_release`?

Reply:
I forgot to add the line